x-pack/plugins/security_solution/public/entity_analytics/components/asset_criticality_file_uploader/components/schedule_risk_engine_callout.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiHorizontalRule,
+  EuiText,
+  EuiFlexItem,
+} from '@elastic/eui';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { i18n } from '@kbn/i18n';
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
+import { formatTimeFromNow } from '../helpers';
+import { useScheduleNowRiskEngineMutation } from '../../../api/hooks/use_schedule_now_risk_engine_mutation';
+import { useRiskEngineStatus } from '../../../api/hooks/use_risk_engine_status';
+
+const TEN_SECONDS = 10000;
+
+export const ScheduleRiskEngineCallout: React.FC = () => {
+  const { data: riskEngineStatus, isLoading: isRiskEngineStatusLoading } =
+    useRiskEngineStatus(TEN_SECONDS);
+  const { addSuccess, addError } = useAppToasts();
+  const { isLoading: isLoadingRiskEngineSchedule, mutate: scheduleRiskEngineMutation } =
+    useScheduleNowRiskEngineMutation({
+      onSuccess: () =>
+        addSuccess(
+          i18n.translate(
+            'xpack.securitySolution.entityAnalytics.assetCriticalityResultStep.riskEngine.successMessage',
+            {
+              defaultMessage: 'Risk engine run scheduled',
+            }
+          )
+        ),
+      onError: (error) =>
+        addError(error, {
+          title: i18n.translate(
+            'xpack.securitySolution.entityAnalytics.assetCriticalityResultStep.riskEngine.errorMessage',
+            {
+              defaultMessage: 'Risk engine schedule failed',
+            }
+          ),
+        }),
+    });
+  const [nextScheduleRun, setNextScheduleRun] = useState<string | undefined>();
```

Comment:
Do we need to use state for this? Cant it be calculated at render time? ... and would that remove the need for the setInterval?

Reply:
The only reason for it to be a state is to trigger a rerender. `useQuery` doesn't trigger a rerender when the API response doesn't change.

Reply:
I just discovered a way of making it work. When passing `structuralSharing: false` to `usuQuery`, it doesn't return the same object, which forces a rerender. 🤯

Reply:
Nice! 😍